server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RecipientRewriteTableProcessor.java

```diff
@@ -147,7 +147,7 @@ private RrtExecutionResult executeRrtForRecipient(Mail mail, MailAddress recipie
         try {
             Mappings mappings = virtualTableStore.getMappings(recipient.getLocalPart(), recipient.getDomain());
 
-            if (mappings != null) {
+            if (mappings != null && !mappings.isEmpty()) {
```

Comment:
still a useless null check?

Reply:
Mocking was hard to change, so I prefered leaving the null check to not complexify the test suite.